daemon/k8s_watcher.go

```diff
@@ -1673,7 +1715,46 @@ func (d *Daemon) deleteK8sPodV1(pod *v1.Pod) {
 	default:
 		logger.Debug("Deleted ipcache map entry on pod delete")
 	}
+}
+
+func (d *Daemon) updateK8sV1Namespace(oldNS, newNS *v1.Namespace) {
+	if oldNS == nil || newNS == nil {
+		return
+	}
 
+	// We only care about label updates
+	if comparator.MapStringEquals(oldNS.GetLabels(), newNS.GetLabels()) {
+		return
+	}
+
+	oldNSLabels := map[string]string{}
+	newNSLabels := map[string]string{}
+
+	for k, v := range oldNS.GetLabels() {
+		oldNSLabels[policy.JoinPath(ciliumio.PodNamespaceMetaLabels, k)] = v
+	}
+	for k, v := range newNS.GetLabels() {
+		newNSLabels[policy.JoinPath(ciliumio.PodNamespaceMetaLabels, k)] = v
+	}
+
+	oldLabels := labels.Map2Labels(oldNSLabels, labels.LabelSourceK8s)
+	newLabels := labels.Map2Labels(newNSLabels, labels.LabelSourceK8s)
+
+	oldIdtyLabels, _ := labels.FilterLabels(oldLabels)
+	newIdtyLabels, _ := labels.FilterLabels(newLabels)
+
+	eps := endpointmanager.GetEndpoints()
+	for _, ep := range eps {
+		epNS := ep.GetK8sNamespace()
+		if oldNS.Name == epNS {
+			err := ep.ModifyIdentityLabels(d, newIdtyLabels, oldIdtyLabels)
```

Comment:
This will add new namespace labels as custom labels on the endpoint which is incorrect. We need a new function which updates the orchestration labels. We also need to ensure that the informational orchestration labels are updated correctly.

Reply:
Discussed offline, the function is already used in the same context. It is still a bug but not specific to this PR.